controllers/discovery.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	bootstrapapi "k8s.io/cluster-bootstrap/token/api"
+	bootstraputil "k8s.io/cluster-bootstrap/token/util"
+	capiremote "sigs.k8s.io/cluster-api/pkg/controller/remote"
+)
+
+const (
+	defaultTokenTTL = 10 * time.Minute
+)
+
+// createTokenInRemoteCluster attempts to create a token with the given ID.
+func createTokenInRemoteCluster(remoteclient capiremote.ClusterClient) (string, error) {
```

Comment:
This interface is too wide, we only really need  to pass in an interface that can run `Create(*v1.Secret)` The caller can set up a remote client and pass in a creator. I would also recommend changing the name to `createToken` since the client is the thing that dictates which cluster this is created in. That detail should not be hard-coded.

Calling this function would look like this:

```
creator := remoteclient.CoreV1().Secrets(metav1.NamespaceSystem)
createToken(creator)
```